todo_app/app.py

```diff
@@ -1,11 +1,25 @@
-from flask import Flask
+from flask import Flask, render_template, request, redirect, url_for
+from todo_app.data.session_items import get_items, add_item
 
 from todo_app.flask_config import Config
 
 app = Flask(__name__)
 app.config.from_object(Config())
 
 
-@app.route('/')
+@app.route('/', methods = ['POST', 'GET'])
 def index():
-    return 'Hello World!'
+    if request.method == 'POST':
+        title = request.form['add-items']
+        add_item(title)
+        # Redirect to the index page to avoid form resubmission on page refresh
+        return redirect(url_for('index'))
+
+    # If it's a GET request or after processing the POST request, render the index page
+    to_do_items = get_items()
+    return render_template('index.html', to_do_items=to_do_items)
```

Comment:
This works nicely, but I'm always a fan of splitting out these two methods - then when/if they get longer it's easier to compartmentalize them

```python
@app.route('/', methods = ['GET'])
def index():
    to_do_items = get_items()
    return render_template('index.html', to_do_items=to_do_items)

@app.route('/', methods=['POST'])
def new_item():
    title = request.form['add-items']
    add_item(title)
    # Redirect to the index page to avoid form resubmission on page refresh
    return redirect(url_for('index'))
```